### ISSUE TITLE
`_format_value` can mangle image names and break plugins

### ISSUE THREAD

**Issue by elevans** [imagej/pyimagej#120]
While investigating #119 we discovered the following bug. In this example we want to open two images, convert them to `ImagePlus` and stitch them with the `Pairwise stitching` plugin.

```python
import imagej
import scyjava as sj

# start fiji
ij = imagej.init('sc.fiji:fiji:2.1.1', headless=False)

# get some classes and services
ImagePlus = sj.jimport('ij.ImagePlus')
ConvertService = ij.get('org.scijava.convert.ConvertService')

# open some images
img1 = ij.io().open('path/to/img1')
img2 = ij.io().open('path/to/img2')

# convert ImgPlus to ImagePlus -- for pairwise stitching plugin
imp1 = ConvertService.convert(img1, ImagePlus)
imp2 = ConvertService.convert(img2, ImagePlus)

ij.py.run_plugin('Pairwise stitching', {"first_image":imp1, "second_image":imp2})
```

This errors out with a syntax error. In my case the images are jpgs named `test_img_1.jpg` and `test_img_2.jpg`.

```

---		---
Error:		')' expected in line 1:
	
		var ; initializeSciJavaParameters ( ) ; run ( "Pairwise stitching" , " first_image=[img[" <test_img_1> . jpg " (-3), 8-...

```

Digging further the offending method appears to be `_format_value` in `pyimagej` which, when passed an image, attempts to stringify the image -- see code block [here](https://github.com/imagej/pyimagej/blob/458c6ca0d4791865f2b55fb991d689feb73bf82d/imagej/__init__.py#L669-L674).

Calling `_format_value` directly on my images returns:

```python
>>>  ij.py._format_value(imp1)
[img["test_img_1.jpg" (-3), 8-bit, 400x400x3x1x1]]
```

This gets passed to `Pairwise stitching` and it complains as it breaks the syntax. One possible fix that needs more work is first checking if the object passed is an ImagePlus and if so returning the `ImagePlus.getID()`.

```python
        def _format_value(self, value):
            if sj.jclass('ij.ImagePlus').isInstance(value):
                return str(value.getID())
            temp_value = str(value).replace('\\', '/')
            if temp_value.startswith('[') and temp_value.endswith(']'):
                    return temp_value
            final_value = '[' + temp_value + ']'
            return final_value
```

**Comment by imagejan** [comment]
`_format_value` just adds the `[]` around the string, which is good to make it safe for strings containing spaces. But the culprit here is `str(value)` already, no?

The IJ1 macro recorder would record image *titles* in this case for the option string. We do have `StringToImagePlusConverter` in `imagej-legacy` that also supports image IDs, but does that work for setting parameters of a `GenericDialog` (see [code](https://github.com/fiji/Stitching/blob/f151a5372c7f445e341c2e95b3ea95f80cde3de5/src/main/java/plugin/Stitching_Pairwise.java#L120-L121)) as well? I don't think so.

**Comment by ctrueden** [comment]
Why not just always call `getTitle()` if the argument is an `ImagePlus`? Then the caller doesn't always have to do it. And if they want the ID, they can call `imp.getID()` explicitly instead.

**Comment by elevans** [comment]
@ctrueden I agree! I tested it and it resolves this issue (commit 13667e52c82b96d6bb69fc60001319020f64881b) . It also allows one trick we discussed with #119 using `show()` and `.getWindow.setVisible(False)` to quickly open and hide the image. A pseudo-headless of sorts.
